modules/ui/fields/sources.js

```diff
@@ -0,0 +1,134 @@
+import { dispatch as d3_dispatch } from 'd3-dispatch';
+import { select as d3_select } from 'd3-selection';
+
+import { t } from '../../core/localizer';
+import { utilGetSetValue, utilNoAuto, utilRebind } from '../../util';
+
+export function uiFieldSources(field, context) {
+    let dispatch = d3_dispatch('change');
+    let items = d3_select(null);
+    let _tags = {};
+    let _selection = d3_select(null);
+    let _pendingChange;
+
+    const mainKey = 'source';
+    const sourceHeader = mainKey + ':';
+
+    // Pre-selected subkey values to show
+    const possibleSourceSubkeys = [{key:'name', value:'Name'}, {key:'url', value:'URL'}, {key:'date', value:'Date'}];
+
+    function scheduleChange() {
+        if (!_pendingChange) return;
+        dispatch.call('change', this, _pendingChange);
+        _pendingChange = null;
+        _selection.call(sources);
+    }
+
+    function valueChange(d3_event, d) {
+        // exit if this is a multiselection and no value was entered
```

Comment:
Make sure to (manually) test this field when multiple features are selected, especially multiple features of various types. Check what happens when none of the selected features has any source tag and when they all have a mixture of different source tags.

Reply:
I created several points, lines, areas, each one having a different feature type and different sources and apparently everything is working as it should. If the features all have the same value on a specific key, they shows up in the raw tag editor as that value. If they have different values, it shows up as "Multiple Values". The values only change if I update one of the values in the field and that value is updated for everything that is selected.